components/director/internal/domain/certsubjectmapping/service.go

```diff
@@ -103,3 +104,14 @@ func (s *service) List(ctx context.Context, pageSize int, cursor string) (*model
 
 	return csmPage, nil
 }
+
+// ListByConsumerID retrieves all certificate subject mappings for a specific consumer id
+func (s *service) ListByConsumerID(ctx context.Context, consumerID string) ([]*model.CertSubjectMapping, error) {
```

Comment:
Can we just delete the records instead of fetching them and deleting them one by one? We could use `r.deleterGlobal.DeleteManyGlobal(ctx, repo.Conditions{repo.NewEqualCondition("internal_consumer_id", consumerID)})`
Then we could just invoke the method in the Application template resolver and this will save us some db requests

Reply:
Of course it will be better ....maybe its misunderstanding...i thought that you want first to list them and then delete them one by one ....